internal/process/provisioning/resolve_creds_hap_selection_migration_test.go

```diff
@@ -0,0 +1,226 @@
+				require.NoError(t, err)
+				require.NotZero(t, n)
+			}
+		} else {
+			if writeFiles {
+				n, err := newImplementationLog.WriteString(originalLabel + "\n")
+				require.NoError(t, err)
+				require.NotZero(t, n)
+			}
+		}
+
+		// count query times
+		if _, ok := queriesCountMap[originalLabel]; !ok {
+			queriesCountMap[originalLabel] = 1
+		} else {
+			if increase {
+				queriesCountMap[originalLabel]++
+			} else {
+				queriesCountMap[originalLabel]--
+			}
+		}
+
+		if _, ok := labelsResultsMap[originalLabel]; !ok {
+
+			// filter out labels that select secrets without that label
+			// for example !dirty selects a secret without dirty label
+			filteredSplit := []string{}
+			split := strings.Split(originalLabel, ",")
+			for _, label := range split {
+				label = strings.Trim(label, " ")
+				if label != "" && label != " " && label != "!dirty" && label != "!euAccess" && label != "shared!=true" && label != "!tenantName" {
+					// at this point all labels should be in format key=value
+					require.Contains(t, label, "=", "Found a signle value label: %s", label)
+					filteredSplit = append(filteredSplit, label)
+				}
+			}
+
+			// secret is queried for the first time - new label, add
+			finalLabels := strings.Join(filteredSplit, ",")
+			labelsResultsMap[originalLabel] = fixDummySecretBinding(finalLabels)
+
+			// add secret so that invoked step can continue its execution
+			unstructuredList.Items = append(unstructuredList.Items, *labelsResultsMap[originalLabel])
+		}
+
+		return true, unstructuredList, nil
+	})
+
+	providers := []pkg.CloudProvider{pkg.AWS,
+		pkg.Azure, pkg.GCP, pkg.SapConvergedCloud,
+	}
+
+	accountProvider := hyperscaler.NewAccountProvider(hyperscaler.NewAccountPool(gc, namespace), hyperscaler.NewSharedGardenerAccountPool(gc, namespace))
+
+	for _, planId := range broker.PlanIDsMapping {
+		for _, platformRegion := range platformRegions {
+			for _, clusterRegion := range clusterRegions {
+				for _, provider := range providers {
+
+					if writeFiles {
+						n, err := oldImplementationLog.WriteString(fmt.Sprintf("INPUT: PlanID: %s, PlatformRegion: %s, ClusterRegion: %s, Provider: %s\n", planId, platformRegion, clusterRegion, provider))
+						require.NoError(t, err)
+						require.NotZero(t, n)
+
+						n, err = newImplementationLog.WriteString(fmt.Sprintf("INPUT: PlanID: %s, PlatformRegion: %s, ClusterRegion: %s, Provider: %s\n", planId, platformRegion, clusterRegion, provider))
+						require.NoError(t, err)
+						require.NotZero(t, n)
+					}
+
+					op := fixOperationWithPlanPlatformRegionAndClusterRegion(planId, platformRegion, clusterRegion, provider)
+					err := memoryStorage.Operations().InsertOperation(op)
+					assert.NoError(t, err)
+					step := NewResolveCredentialsStep(memoryStorage.Operations(), accountProvider, &rules.RulesService{})
+
+					// first run
+					increase = true
+					_, when, err := step.Run(op, fixLogger())
```

Comment:
So here we have reactor called and queriesCountMap filled.
Comparison is based on assumption that map key will be the same (`GetListRestrictions().Labels.String()`).
What if order is not stable?
What if there is irrelevant label added?
Shouldn't we compare explicitely shared, euaccess, hyperscalertype labels?